src/input_map.rs

```diff
@@ -200,51 +215,22 @@ impl<A: Actionlike> InputMap<A> {
         self
     }
 
-    /// Inserts multiple action-input bindings.
-    /// Note that the type of your iterators must be homogeneous.
+    /// Inserts multiple action-input bindings provided by an iterator.
+    /// Note that all elements within the iterator must be of the same type (homogeneous).
     ///
     /// This method ensures idempotence, meaning that adding the same input
     /// for the same action multiple times will only result in a single binding being created.
     #[inline(always)]
     pub fn insert_multiple(
         &mut self,
-        bindings: impl IntoIterator<Item = (A, impl Into<UserInput>)>,
+        bindings: impl IntoIterator<Item = (A, impl UserInput)>,
     ) -> &mut Self {
         for (action, input) in bindings.into_iter() {
             self.insert(action, input);
         }
         self
     }
 
-    /// Insert a mapping between the simultaneous combination of `buttons` and the `action` provided
-    ///
-    /// Any iterator convertible into a [`InputKind`] can be supplied,
-    /// but will be converted into a [`HashSet`](bevy::utils::HashSet) for storage and use.
-    /// Chords can also be added with the [insert](Self::insert) method if the [`UserInput::Chord`] variant is constructed explicitly.
-    ///
-    /// When working with keyboard modifier keys, consider using the `insert_modified` method instead.
-    pub fn insert_chord(
```

Comment:
It would be nice to keep these two convenience methods if possible.

Reply:
I think it might be a bit inconvenient because if you use impl IntoIterator<Item = impl UserInput> as a parameter, you know that the iterated elements must be of the same type due to the limitations of generics